### PR TITLE
fix: handle non-AIMessage instances in agent.run() output detection

### LINKED ISSUE #446: [TS] No output generated
In the latest of mcp-use@latest (1.2.3) when any request is sent even the simpliest one result always is 'No output generated'

```
const result = await agent.run('hello')

console.log(result ) //result - 'No output generated'
```

### PR DESCRIPTION
**Pull request by gabrielekarra**
## Description

Fixes #446 where `agent.run()` returned `'No output generated'` even when the agent produced valid output.

## Problem

The agent was only accepting messages where `message instanceof AIMessage` was true. However, in real usage, the LangChain agent graph often emits messages as plain JavaScript objects like `{ type: 'ai', content: '...' }` due to:
- Version mismatches between `@langchain/core` packages
- Serialization/deserialization across module boundaries  
- Different message formats from various LangChain versions

## Solution

Added two production-ready helper methods to detect assistant-like messages regardless of their class:

1. **`_isAIMessageLike()`** - Detects assistant messages by:
   - Checking for real `AIMessage` instances (backwards compatible)
   - Accepting objects with `type: 'ai'` or `type: 'assistant'`
   - Accepting objects with `role: 'ai'` or `role: 'assistant'`  
   - Checking for `getType()` and `_getType()` methods
   - Requiring `content` property to be present

2. **`_messageHasToolCalls()`** - Safely checks for tool_calls array presence

## Testing

- Created comprehensive test suite with 9 test cases covering all edge cases
- All existing integration tests pass (16 tests in 7 files)
- 100% backwards compatible with existing `AIMessage` instances
- Tested with multiple message formats and LangChain versions

## Changes

- Modified `src/agents/mcp_agent.ts` to use the new helper methods
- No breaking changes
- Minimal code change with maximum compatibility

Closes #446

### CODE REVIEW

File: libraries/typescript/packages/mcp-use/src/agents/mcp_agent.ts [hunk 1]
```diff
@@ -669,6 +669,80 @@ export class MCPAgent {
...
+    
+    // Try methods first (for partially deserialized objects)
+    if (typeof msg.getType === "function") {
+      const type = msg.getType();
+      return type === "ai" || type === "assistant";
+    }
+    if (typeof msg._getType === "function") {
+      const type = msg._getType();
+      return type === "ai" || type === "assistant";
+    }
```
Comment (harshithvh) [inline libraries/typescript/packages/mcp-use/src/agents/mcp_agent.ts:716]: Consider wrapping in try-catch blocks with a fallback

**Comment by tonxxd** [comment]
@claude

**Comment by tonxxd** [comment]
Thanks @gabrielekarra I started from your pr and made some changes. It's merged in canary and will be released in main soon. Closing this pr in favor of https://github.com/mcp-use/mcp-use/pull/453
